contracts/voting/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+#![no_std]
+
+use soroban_sdk::{
+    contracterror, contractimpl, contracttype, vec, Address, ConversionError, Env, Map, Symbol,
+};
+
+#[contracterror]
+#[derive(Clone, Debug, Copy, Eq, PartialEq, PartialOrd, Ord)]
+#[repr(u32)]
+pub enum Error {
+    Conversion = 1,
+    KeyExpected = 2,
+    NotFound = 3,
+    AlreadyVoted = 4,
+    DuplicatedEntity = 5,
+    Overflow = 6,
+    VotingClosed = 7,
+    NotValidID = 8,
+}
+
+impl From<ConversionError> for Error {
+    fn from(_: ConversionError) -> Self {
+        Error::Conversion
+    }
+}
+
+#[contracttype]
+#[derive(Clone, Copy)]
+pub enum DataKey {
+    Admin = 0,
+    VoterList = 1,
+    Proposals = 2,
+}
+
+pub struct ProposalVotingContract;
+
+#[contractimpl]
+impl ProposalVotingContract {
+    pub fn init(env: Env, admin: Address) {
+        env.storage().set(&DataKey::Admin, &admin);
+        env.storage()
+            .set(&DataKey::Proposals, &Map::<u64, Proposal>::new(&env));
+    }
+
+    pub fn create_proposal(env: Env, id: u64) -> Result<(), Error> {
```

Comment:
Currently its an ID .. but maybe adding a hash field would be better. Let me know your thoughts.

Reply:
You said this is not doable in init?

Reply:
SOrry missed this reply, i replied here : https://github.com/eigerco/nebula/pull/12#issuecomment-1655328376